src/components/AsideHeader/AsideHeader.tsx

```diff
@@ -1,32 +1,59 @@
 import React, {useMemo} from 'react';
+import debounceFn from 'lodash/debounce';
 
 import {Content} from '../Content';
 
 import {AsideHeaderContextProvider, AsideHeaderInnerContextProvider} from './AsideHeaderContext';
 import {AsideHeaderProps} from './types';
 
-import {FirstPanel} from './components';
+import {FirstPanel, TopPanel} from './components';
 import {b} from './utils';
 
 import './AsideHeader.scss';
 import {ASIDE_HEADER_COMPACT_WIDTH, ASIDE_HEADER_EXPANDED_WIDTH} from '../constants';
 import {useAsideHeaderInnerContextValue} from './useAsideHeaderInnerContextValue';
 
 export const AsideHeader = (props: AsideHeaderProps) => {
-    const {className, compact} = props;
+    const {className, compact, topAlert} = props;
     const size = compact ? ASIDE_HEADER_COMPACT_WIDTH : ASIDE_HEADER_EXPANDED_WIDTH;
     const asideHeaderContextValue = useMemo(() => ({size, compact}), [compact, size]);
     const asideHeaderInnerContextValue = useAsideHeaderInnerContextValue({...props, size});
+
+    const topRef = React.useRef<HTMLDivElement>(null);
+
+    const [maxHeightWithTop, setMaxHeightWithTop] = React.useState<string | undefined>();
+    const topHeight = topRef.current ? topRef.current.clientHeight : 0;
+
+    React.useLayoutEffect(() => {
+        const updateTopSize = debounceFn(
+            () => {
+                if (topRef.current) {
+                    setMaxHeightWithTop(`calc(100vh - ${topRef.current.clientHeight}px)`);
+                }
+            },
+            200,
+            {leading: true},
+        );
+        if (topAlert) {
+            window.addEventListener('resize', updateTopSize);
+            updateTopSize();
+        }
+        return () => window.removeEventListener('resize', updateTopSize);
+    }, [topAlert, topRef, topHeight]);
+
     return (
         <AsideHeaderContextProvider value={asideHeaderContextValue}>
             <AsideHeaderInnerContextProvider value={asideHeaderInnerContextValue}>
                 <div className={b({compact}, className)}>
+                    {/* Top Panel */}
+                    <TopPanel ref={topRef} />
```

Comment:
Can we not render the component `topAlert` is not set?

Reply:
fixed